### DISCUSSION TITLE
"Sommerpause" beim Gaszähler und MQTT Daten

### DISCUSSION THREAD

**Discussion by AdmiralRaccoon** [jomjol/AI-on-the-edge-device#3065] [General]
Hallo,
Mir wurde letztes Jahr schon mitgeteilt das man den AOTE Device ja auch einfach vom Strom nehmen kann, wenn sich am Gaszähler eh nichts tut, da er dann ab dem nächsten Herbst / Winter wieder seinen Dienst aufnehmen kann.

Hat jemand einen Idee, wie ich trotzdem den letzten Wert für die Entität im Home Assistant beibehalten kann, da ja jetzt alle Sensoren und Utility Meter auf "unavailable" gesprungen sind?

Danke & Gruss,
Philipp

<img width="497" alt="image" src="https://github.com/jomjol/AI-on-the-edge-device/assets/41727850/9d806bed-b6f1-43d9-b3c9-9ce6bc892cdf">

**Comment by caco3** [comment]
Das Discovery ist so konfiguriert, dass der available/unavailable Status über das LWT (Last Will Topic) gesteuert wird.
Und das LWT ist so konfiguriert, dass der Broker das device auf unavailable setzt, wenn die Verbindung für eine bestimmte Zeit unterbrochen ist. Diese Zeit ist auf `roundInterval * 60 * 2.5` Sekunden festgelegt (Wird auch im DEBUG Log ausgegeben).

Du kannst diesen Status übersteuern, indem du mit einem beliebigen MQTT-Client entweder `connected` resp. `connection lost` ins  Topic `<MAINTOPIC>/connection` schreibst.

Das funktioniert auch bei einem device, das noch verbunden wird, wobei das Topic periodisch mit `connected` überschrieben wird.

Wenn Du das Topic eines ausgeschalteten devices auf `connected` setzt, sollten die letzten Werte wieder angezeigt werden. ich weis jedoch nicht, ob das zyklisch gemacht werden muss oder obs einmalig reicht.

**Reply by caco3** [reply]
Oder du könntest auch jedes relevante Discovery topic umschreiben:
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/1783586/89797ad2-dfcd-4a16-bd6b-bcc50abf724e)

Würde ich aber nicht empfehlen, da es  unbekannte Nebeneffekte haben könnte. Beim Neustart des devices wird das aber wieder korrigiert.

**Reply by AdmiralRaccoon** [reply]
Danke für die Tipps....aber wurschtel ich iieber nicht drin rum.

Ich kann ja das Device auch quasi laufen lassen, und z.b. nur 1x am Tag eine Lesung machen.
